packages/auth/src/authClients/firebase.ts

```diff
@@ -1,11 +1,28 @@
-import type FirebaseNamespace from 'firebase/app'
+import {
+  isSignInWithEmailLink,
+  sendSignInLinkToEmail,
+  signInWithEmailAndPassword,
+  createUserWithEmailAndPassword,
+  OAuthProvider,
+  signInWithEmailLink,
+  signInWithPopup,
+} from '@firebase/auth'
+import type {
+  ActionCodeSettings,
+  User,
+  CustomParameters,
+  Auth as FirebaseAuth,
+} from '@firebase/auth'
 
 import { AuthClient } from './'
 
-export type Firebase = typeof FirebaseNamespace
+export type FirebaseUser = User
+export type Firebase = FirebaseAuth
+export type emailLinkProvider = 'emailLink'
 
 // @TODO: Firebase doesn't export a list of providerIds they use
 // But I found them here: https://github.com/firebase/firebase-js-sdk/blob/a5768b0aa7d7ce732279931aa436e988c9f36487/packages/rules-unit-testing/src/api/index.ts
+// NOTE: 2021-09-07 could we replace this with ProviderId type now exported from firebase-sdk-js ^v9.0.0
```

Comment:
This wasn't available prior to v-9, we can probably now remove this and import from firebase

Reply:
I tried to use the firebase exported const/enum-like object ProviderId but ran into a bit of typescript hurdles. Probably not super important to make this change at the moment, since it works fine as is.